src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,302 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.MediaFiles.TorrentInfo;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.RemotePathMappings;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Validation;
+using System.Net;
+using static NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationTask;
+using System.IO;
+
+namespace NzbDrone.Core.Download.Clients.DownloadStation
+{
+    public class DownloadStation : TorrentClientBase<DownloadStationSettings>
+    {
+        protected readonly IDownloadStationProxy _proxy;
+        protected static readonly Dictionary<DownloadStationTaskStatus, DownloadItemStatus> StatusTable = new Dictionary<DownloadStationTaskStatus, DownloadItemStatus>();
+
+        static DownloadStation()
+        {
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Waiting, DownloadItemStatus.Queued);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Error, DownloadItemStatus.Failed);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Paused, DownloadItemStatus.Paused);
+            StatusTable.Add(DownloadStationTask.DownloadStationTaskStatus.Seeding, DownloadItemStatus.Completed);
+        }
+
+        public DownloadStation(IDownloadStationProxy proxy,
+                     ITorrentFileInfoReader torrentFileInfoReader,
+                     IHttpClient httpClient,
+                     IConfigService configService,
+                     IDiskProvider diskProvider,
+                     IRemotePathMappingService remotePathMappingService,
+                     Logger logger)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            this._proxy = proxy;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "Download Station";
+            }
+        }
+
+        protected virtual OsPath GetOutputPath(OsPath outputPath, DownloadStationTask torrent)
+        {
+            return outputPath + torrent.Title;
+        }
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            IEnumerable<DownloadStationTask> torrents;
+
+            try
+            {
+                torrents = _proxy.GetItems(Settings);
+            }
+            catch (DownloadClientException ex)
+            {
+                _logger.Error(ex, ex.Message);
+                return Enumerable.Empty<DownloadClientItem>();
+            }
+
+            var items = new List<DownloadClientItem>();
+
+            foreach (var torrent in torrents)
+            {
+                var outputPath = new OsPath(torrent.Additional.Detail["destination"]);
```

Comment:
1) You already have DownloadStationFixture , which correctly inherit from DownloadClientFixtureBase<TClient>, look at existing subclasses for inspiration, TransmissionFixtureBase is probably pretty close.

2) For filestation you might want to create a second IFileStationProxy (api) and ISharedFolderResolver (service+caching), considering it's a different service on the Syno. IDownloadStationProxy instead of IFileStationProxy might be okay too, I think, but separate class is cleaner. @markus101  may have an opinion.

3) Place it in the *.Clients.DownloadStation namespace.

4) Check out ICacheManager (factory) and ICached, there are quite a few places where caching is used. In your case I'd use ICached<string> so you can store and track individual mappings.

5) Good question, coz this is tricky. We can add downloads to DownloadStation, for that we'd use the Shared Folder path, not the filesystem path.
The GetItems function should first filter by Category/TvDirectory.
Then it runs GetOutputPath, which should resolve the Shared Folder.
Finally RemotePathMapping is applied.
Probably an idea to write a list/table of all scenarios.